Python/pystrhex.c

```diff
@@ -57,7 +57,7 @@ static PyObject *_Py_strhex_impl(const char* argbuf, const Py_ssize_t arglen,
     }
     resultlen += arglen * 2;
 
-    if (abs_bytes_per_sep >= arglen) {
+    if (abs_bytes_per_sep >= (unsigned int) arglen) {
```

Comment:
`arglen` can be larger than the maximum of `unsigned int`.
```suggestion
    if ((size_t)abs_bytes_per_sep >= (size_t)arglen) {
```

Reply:
Okay.